hnn_core/dipole.py

```diff
@@ -137,9 +137,29 @@ def simulate_dipole(net, n_trials=1, n_jobs=1):
     dpl: list | instance of Dipole
         The dipole object or list of dipole objects if n_trials > 1
     """
-    parallel, myfunc = _parallel_func(_clone_and_simulate, n_jobs=n_jobs)
-    out = parallel(myfunc(net.params, idx) for idx in range(n_trials))
-    dpl, spiketimes, spikegids = zip(*out)
+
+    from .parallel import create_parallel_context, get_nhosts
+
+    if n_jobs > 1:
+        # check whether NEURON is using parallel nrniv processes
+        create_parallel_context(n_cores=1)
```

Comment:
even though n_cores=1 ends up calling ParallelContext(1), that does not guarantee that MPI is not being used (nhosts=1). Running the simulation via nrniv will set nhosts to whatever the number of processors were specified on the command line.